exasol/analytics/query_handler/query/select.py

```diff
@@ -44,30 +50,41 @@ def audit_fields(self) -> dict[str, Any]:
         return self._audit_fields
 
 
-class AuditQuery(AuditData):
+class AuditQuery(SelectQueryWithColumnDefinition, AuditData):
     """
-    A query wrapper for a special read-only query that selects data for auditing.
+    A wrapper for a special read-only query that selects data for auditing. An object
+    of the class can also be used as an audit property bag, since it inherits from the
+    `AuditData` as well as `SelectQueryWithColumnDefinition`. The query is optional.
+    If provided the output columns should also be defined.
     """
-    def __init__(self, query_with_columns: SelectQueryWithColumnDefinition,
+    def __init__(self, query_string: str = '',
+                 output_columns: list[Column] | None = None,
```

Comment:
if output columns can get None here, we need to adapt the type hints in SelectQueryWithColumnDefinition, what speaks against it to have SelectQueryWithColumnDefinition as an optional attribute and inherit from Query instead and do a simple implementation for query_text property.

Reply:
This is the question of whether the AuditQuery IS Query or ENCAPSULATES a query. If it IS a Query, which should be assumed if the class inherits from `Query` then the way I wrote it is the correct way of doing this. The logic suffers a bit from the fact that it is now optional. Maybe setting the default value to "SELECT 1" will work better.

Reply:
So that will be my proposal: Remove the validation, replace the empty string with "SELECT 1", so that we can always run the query from pyexasol - no checks needed. The columns should define the columns we collect from the results. If it's an empty list - fair game. I might have needed to execute a query for some reason but not collect anything for auditing. I can't think of any use case right now, but who knows?

Reply:
I think, your proposal has one problem, and this makes it hard for us, to detect whether the query is intended or not to be logged.
if we have instead a separate select_with_column property, we know, if it is None, that we don't need it and regarding the implementation of the query_text property in the AuditQuery, we return either the query_text of te select_with_columns or SELECT 1